ipmininet/router/config/zebra.py

```diff
@@ -83,6 +86,59 @@ def has_started(self):
         # We override this such that we wait until we have the API socket
         return os.path.exists(self.zebra_socket)
 
+    @classmethod
+    def incr_last_routerid(cls):
+        cls.last_routerid += 1
+
+    def _equal_routerid(self, n):
+        # Check that the most-visible IPv4 address is not in conflict
+        # with the current router id
+        ip_list = sorted((ip for itf in realIntfList(n)
+                          for ip in itf.ips()),
+                         cmp=address_comparator)
+        if len(ip_list) != 0 \
+                and int(ip_list.pop().ip) == int(self.last_routerid):
+            return True
+
+        # Check that a router id explicitly set
+        # in any other Quagga daemon is not in conflict
+        # with the current router id
+        for d in n.config.daemons:
+            if issubclass(d, QuaggaDaemon) and d != self \
+                    and hasattr(d, "routerid") \
+                    and int(d.routerid) == int(self.last_routerid):
+                return True
+        return False
+
+    def get_routerid(self):
+        """Computes the default router id for Quagga daemons
+        It returns the most-visible IPv4 address among its router interfaces.
+        If no IPv4 addresses exists for its router, it generates a unique router id"""
+        ip_list = sorted((ip for itf in realIntfList(self._node)
```

Comment:
The generated router id should be the same for all daemons not manually overwriting it for consistency reasons (i.e. not having to remember a router by 4 different identifiers ...)